### PR TITLE
Support custom PROJ datum shifts

### PR DESCRIPTION
**Pull request by PeterPetrik**
Description of the PROJ initialization and workflow is in inutprojutils.h file

### CODE REVIEW

**Comment by saberraz** [comment]
I am having serious issue with the builds from this PR:

1- Create a gpkg with EPSG:27700
2- Save the project and try to load it in Input
3- Crash

Tested on Android 10.

**Comment by PeterPetrik** [comment]
> I am having serious issue with the builds from this PR:
> 
> 1- Create a gpkg with EPSG:27700
> 2- Save the project and try to load it in Input
> 3- Crash
> 
> Tested on Android 10.

probably related to https://github.com/qgis/QGIS/pull/40482

**Comment by saberraz** [comment]
@PeterPetrik not really. There is no background layer. Just a single gpkg layer:
https://public.cloudmergin.com/projects/saber/proj_issue/tree

**Review by tomasMizera — APPROVED** [review]
Thanks for personal explanation. I have some notes, but other than that, it looks good! 

 - there are several `qDebugs` instead of inputUtils::log please check if they are intended (more than just the one I pointed at)
 - developer documentation needs to be updated that we now need two new variables defined in config
 - documentation on help.inputapp.io could tell that the easiest solution with proj is to upload them to Mergin together with project. So far it guides users to copy proj files manually to device, however It will be harder to copy files to android devices when we will target Android 11 due to a scoped storage.

**Comment by tomasMizera** [comment]
And also changes need to be merged to QGIS before merging this PR

File: app/inputprojutils.cpp [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+    variable = true;
+  }
+}
+
+void InputProjUtils::cleanCustomDir()
+{
+  QDir dir( mCurrentCustomProjDir );
+  if ( !dir.isEmpty() )
+  {
+    qDebug() << "InputPROJ: cleaning custom proj dir " << mCurrentCustomProjDir;
```
[the remark this answers is not in the document]
Comment (PeterPetrik) [reply app/inputprojutils.cpp:50]: grep shows we have `qDebug()` all over the place. so i would keep it like that

[the remark this answers is not in the document]
Comment (tomasMizera) [reply app/inputprojutils.cpp:50]: sure, I just though that we want to see it in user logs when they submit it.

[the remark this answers is not in the document]
Comment (PeterPetrik) [reply app/inputprojutils.cpp:50]: i do not think it has any relevance to the user (log)...

**Comment by PeterPetrik** [comment]
created QGIS PR, updated help docs

**Comment by PeterPetrik** [comment]
Change: we copy all custom proj files to specified folder on Input startup. So if the project contains a custom proj files, application needs to be restarted for them to be properly loaded. Changed manual test & docs accordingly

**Pull request merged**